libminifi/include/io/Serializable.h

```diff
@@ -22,11 +22,36 @@
 #include <string>
 #include "EndianCheck.h"
 #include "DataStream.h"
+
+namespace {
+  template<typename Integral, typename std::enable_if<
```

Comment:
Far less readable and godbolt and my i7 showed no benefit to this. It seems like syntactical candy. Profiling shows no benefit whatsoever. Having functions with specific functors makes it easy on the development side to specifically type what we set. Your ticket "Type-unsafe template functions" I disagree. I think this is more type unsafe and less easy for developers to read. 

"Code duplication" I think this is true, but it results in perhaps some easier development for some. I think we should keep this as is and what was added is syntactical sugar with the potential negative that it can be more confusing during development.

Reply:
You are right in the part that this doesn't improve either performance or readibility.
This change became useful in the template write(Integral) function, where this can be called easily, so I don't need to copy-paste writeData calls and could remove the local template writeData implementation. 
Overall I see more gain here.